src/WP_CLI/SearchReplacer.php

```diff
@@ -114,7 +116,13 @@ private function _run( $data, $serialised, $recursion_level = 0, $visited_data =
 					$search_regex .= $this->regex_delimiter;
 					$search_regex .= $this->regex_flags;
 
-					$result = preg_replace( $search_regex, $this->to, $data, $this->regex_limit );
+					if ( $this->callback ) {
+						if ( strpos( $data, $this->from ) !== false ) {
```

Comment:
The way this logic is expressed, it will throw a bogus PCRE error if the `strpos()` fails.